### PR TITLE
enhanced cached blocks in pos fee estimator

### PR DESCRIPTION
**Pull request by lazynina**

### CODE REVIEW

**Comment by lazynina** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/deso-protocol/core/1290?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#1290** <a href="https://app.graphite.dev/github/pr/deso-protocol/core/1290?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#1289** <a href="https://app.graphite.dev/github/pr/deso-protocol/core/1289?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `feature/proof-of-stake`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @lazynina and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: lib/pos_fee_estimator.go [hunk 1]
```diff
@@ -234,40 +273,40 @@ func (posFeeEstimator *PoSFeeEstimator) UpdateGlobalParams(globalParams *GlobalP
...
+			if ii.Block.Header.GetView() != jj.Block.Header.GetView() {
+				return ii.Block.Header.GetView() < jj.Block.Header.GetView()
 			}
-			iiHash, err := ii.Hash()
+			iiHash, err := ii.Block.Hash()
 			if iiHash == nil || err != nil {
 				return false
 			}
-			jjHash, err := jj.Hash()
+			jjHash, err := jj.Block.Hash()
```
Comment (tholonious) [inline lib/pos_fee_estimator.go:309]:
```suggestion
			jjHash, err := jj.Hash
```

File: lib/pos_fee_estimator.go [hunk 2]
```diff
@@ -234,40 +273,40 @@ func (posFeeEstimator *PoSFeeEstimator) UpdateGlobalParams(globalParams *GlobalP
...
+			if ii.Block.Header.Height != jj.Block.Header.Height {
+				return ii.Block.Header.Height < jj.Block.Header.Height
 			}
-			if ii.Header.GetView() != jj.Header.GetView() {
-				return ii.Header.GetView() < jj.Header.GetView()
+			if ii.Block.Header.GetView() != jj.Block.Header.GetView() {
+				return ii.Block.Header.GetView() < jj.Block.Header.GetView()
 			}
-			iiHash, err := ii.Hash()
+			iiHash, err := ii.Block.Hash()
```
Comment (tholonious) [inline lib/pos_fee_estimator.go:305]:
```suggestion
			iiHash, err := ii.Hash
```

**Review by tholonious — APPROVED** [review]

**Comment by lazynina** [comment]
### Merge activity

* **May 8, 2:31 PM EDT**: @lazynina started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/deso-protocol/core/1290).

**Pull request merged**